appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/policy/FaultTolerancePolicy.java

```diff
@@ -579,15 +601,28 @@ private Object processBulkheadStage(FaultToleranceInvocation invocation) throws
                     try {
                         logger.log(Level.FINE, "Entered bulkhead execution.");
                         // ok, lets run
-                        return proceed(invocation);
+                        Object res = proceed(invocation);
+                        if (!exitOnCompletion) {
+                            return res;
+                        }
+                        return ((CompletionStage<?>) res).whenComplete((value, excetion) -> {
```

Comment:
```suggestion
                        return ((CompletionStage<?>) res).whenComplete((value, exception) -> {
```
Presumably meant to be exception?